sdk/endpoint/endpoint.go

```diff
@@ -180,8 +208,26 @@ func (nsme *nsmEndpoint) Close(ctx context.Context, incomingConnection *connecti
 	return &empty.Empty{}, nil
 }
 
-// NewNSMEndpoint creates a new NSM endpoint
+// NewNSMEndpoint creates a new NSM endpoint with a single registration.
 func NewNSMEndpoint(ctx context.Context, configuration *common.NSConfiguration, service networkservice.NetworkServiceServer) (NsmEndpoint, error) {
+	registration := []Registration{MakeNsmEndpointRegistration(configuration)}
+	return NewNSMEndpointWithRegistrations(ctx, configuration, service, registration)
+}
```

Comment:
NOTE: you can just use this construction
```suggestion
func NewNSMEndpoint(ctx context.Context, configuration *common.NSConfiguration, service networkservice.NetworkServiceServer, registrations...Registartion) (NsmEndpoint, error) {
	if len(registrations) == 0 {
		registrations := []Registration{MakeNsmEndpointRegistration(configuration)}
	}
	return NewNSMEndpointWithRegistrations(ctx, configuration, service, registration) //TODO inline
}
```

Reply:
I think that having an (additional) function that takes an array is more clear than a variadic function with an implicit `len == 0` case.

Reply:
Please take a look at the go `option` pattern. I think actually we have a structure that has a default value of registrations `[]Registration{MakeNsmEndpointRegistration(configuration)}`